torch/autograd/gradcheck.py

```diff
@@ -225,7 +225,18 @@ def fail_test(msg):
             'gradcheck expects at least one input tensor to require gradient, '
             'but none of the them have requires_grad=True.')
 
-    output = _differentiable_outputs(func(*tupled_inputs))
+    func_out = func(*tupled_inputs)
+    output = _differentiable_outputs(func_out)
+
+    if not output:
+        for i, o in enumerate(func_out):
```

Comment:
what about to @SsnL's point that this is only valid if all of the inputs require_grad?

Reply:
I think @ifedan should filter out inputs that don't require grad in lines 233~235.

Reply:
I'm a little concerned that would also miss cases we want to check.  I'd probably just start by asserting that if there are no differentiable outputs and at least one requires_grad input, then all the inputs requires_grad.  Then we could loosen the requirements later if there are cases that run afoul of that check.

Reply:
Added check that if any input requires grad than all the inputs must require grad